### ISSUE TITLE
ScopeDownStatement not supported in ManagedRuleGroupStatement

### ISSUE THREAD

**Issue by abdallah** [umotif-public/terraform-aws-waf-webaclv2#37]
**What is the current behavior?**
Currently `aws_wafv2_web_acl` does not support adding a `scope_down_statement` inside a `managed_rule_group_statement`

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
here's the JSON exported from AWS WAFv2 directly:
```json
{
  "Name": "Amazon-AnonymousIpList",
  "Priority": 1,
  "Statement": {
    "ManagedRuleGroupStatement": {
      "VendorName": "AWS",
      "Name": "AWSManagedRulesAnonymousIpList",
      "ScopeDownStatement": {
        "ByteMatchStatement": {
          "SearchString": "my.example.com",
          "FieldToMatch": {
            "SingleHeader": {
              "Name": "host"
            }
          },
          "TextTransformations": [
            {
              "Priority": 0,
              "Type": "NONE"
            }
          ],
          "PositionalConstraint": "EXACTLY"
        }
      }
    }
  },
  "OverrideAction": {
    "Count": {}
  },
  "VisibilityConfig": {
    "SampledRequestsEnabled": true,
    "CloudWatchMetricsEnabled": true,
    "MetricName": "AmazonAnonymousIpList"
  }
}
```

Translating that to terraform:
```
rule {
    name     = "Amazon-AnonymousIpList"
    priority = 1
    override_action {
      count {}
    }
    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAnonymousIpList"
        vendor_name = "AWS"
        scope_down_statement {
          byte_match_statement {
            search_string = "my.example.com"
            field_to_match {
              single_header {
                name = "host"
              }
            }
            text_transformation {
              priority = 0
              type     = "NONE"
            }
            positional_constraint = "EXACTLY"
          }
        }
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "AmazonAnonymousIpList"
      sampled_requests_enabled   = true
    }
  }
```

The above fails with the following error:
```
❯ terraform validate
╷
│ Error: Unsupported block type
│ 
│   on example.tf line 186, in resource "aws_wafv2_web_acl" "web-acl":
│  186:         scope_down_statement {
│ 
│ Blocks of type "scope_down_statement" are not expected here.
```

**What is the expected behavior?**
according to https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/wafv2_web_acl#managed-rule-group-statement , the managed_rule_group_statement block does not support a `scope_down_statement`. Could you please add that?

**Software versions?**
* Terraform v1.0.0
* hashicorp/aws v3.49.0

**Comment by marcincuber** [comment]
If the `managed_rule_group_statement` block does not support a `scope_down_statement` in the AWS provider, there is nothing we can do here. Once it is supported in Terraform AWS provider please create a new issue.

**Comment by marcincuber** [comment]
Reopening as this functionality has become available in the [3.50.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v3.50.0) release.
@Ohid25 @seanpascual
